bindings/src/test/java/io/pravega/storage/extendeds3/ExtendedS3TestContext.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.storage.extendeds3;
+
+import com.emc.object.s3.S3Config;
+import com.emc.object.s3.bean.ObjectKey;
+import com.emc.object.s3.jersey.S3JerseyClient;
+import com.emc.object.s3.request.DeleteObjectsRequest;
+import com.emc.object.util.ConfigUri;
+import io.pravega.test.common.TestUtils;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Test context Extended S3 tests.
+ */
+public class ExtendedS3TestContext {
+    public static final String BUCKET_NAME_PREFIX = "pravegatest-";
+    public final ExtendedS3StorageConfig adapterConfig;
+    public final S3JerseyClient client;
+    public final S3ImplBase s3Proxy;
+    public final int port = TestUtils.getAvailableListenPort();
+    public final String configUri = "http://127.0.0.1:" + port + "?identity=x&secretKey=x";
+    public final S3Config s3Config;
+
+    public ExtendedS3TestContext() throws Exception {
+        String bucketName = BUCKET_NAME_PREFIX + UUID.randomUUID().toString();
+        this.adapterConfig = ExtendedS3StorageConfig.builder()
+                .with(ExtendedS3StorageConfig.CONFIGURI, configUri)
+                .with(ExtendedS3StorageConfig.BUCKET, bucketName)
+                .with(ExtendedS3StorageConfig.PREFIX, "samplePrefix")
+                .build();
+        s3Config = new ConfigUri<>(S3Config.class).parseUri(configUri);
+        s3Proxy = new S3ProxyImpl(configUri, s3Config);
+        s3Proxy.start();
```

Comment:
There is a chance that something in the constructor fails. We want to put a try-catch block here to shut down the s3proxy if we fail before we exit the constructor.

Reply:
The close is called in  @After in test classes.

Reply:
You will have nothing to call `close` on since this is the constructor. If the constructor throws, there will be no object to call `close` on, but the side effects will remain. It is your responsibility to clean up any such side effects in the constructor should it fail.

Reply:
fixed